addons/survey/tests/test_survey_flow.py

```diff
@@ -16,12 +16,16 @@ def _format_submission_data(self, page, answer_data, additional_post_data):
             if question.question_type == 'multiple_choice':
                 values = answer_vals['value']
                 for value in values:
-                    key = "%s_%s_%s" % (page.survey_id.id, question.id, value)
-                    post_data[key] = value
+                    if question.id in post_data:
```

Comment:
I've seen this multiple times so far.
Maybe introduce a common method in `SurveyCase` ?